### PR TITLE
chore: Use standardized test timeouts and delays

### PR DESCRIPTION
**Pull request by mafredri**
This PR introduces a new package called `testutil` for exposing standard delays and intervals that can be used in tests.

This is so that we can increase or decrease the values depending on platform.

The motivation behind this PR is that Windows tests are flaking due to timeouts, not that they failed to complete.

Motivation behind putting this package in `internal` can be found in #1405.

PS. I'm not married to the name, I tried floating a few different ideas in my mind:

```
codertest
ctest
durtest
testdur
tests
testtime
testwait
testy
timetest
waittest
```

But I didn't really like any of them. I consider this package something we will expand in the future.

- chore: Standardize waits and intervals between tests
- chore: Add linter for timeouts

### CODE REVIEW

File: scripts/rules.go [hunk 1]
```diff
@@ -81,6 +81,45 @@ func doNotCallTFailNowInsideGoroutine(m dsl.Matcher) {
 		Report("Do not call functions that may call t.FailNow in a goroutine, as this can cause data races (see testing.go:834)")
 }
 
+// useStandardTimeoutsAndDelaysInTests ensures all tests use common
+// constants for timeouts and delays in usual scenarios, this allows us
+// to tweak them based on platform (important to avoid CI flakes).
+//nolint:unused,deadcode,varnamelen
+func useStandardTimeoutsAndDelaysInTests(m dsl.Matcher) {
```
Comment (johnstcn) [inline scripts/rules.go:88]: 👍 Nice!

File: internal/testutil/duration.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+package testutil
+
+import "time"
+
+// Constants for timing out operations, usable for creating contexts
+// that timeout or in require.Eventually.
+const (
+	WaitShort = 5 * time.Second
+	WaitLong  = 15 * time.Second
```
Comment (johnstcn) [inline internal/testutil/duration.go:11]: nit: should we have a `WaitMedium` as well for consistency with `Interval`?

**Review by johnstcn — APPROVED** [review]

File: internal/testutil/duration.go (hunk 2 above)
Comment (mafredri) [reply internal/testutil/duration.go:11]: Good suggestion; should we rename Short to Medium with its current value and add a new one that’s even shorter?

File: pty/ptytest/ptytest.go [hunk 3]
```diff
@@ -85,7 +86,7 @@ type PTY struct {
 func (p *PTY) ExpectMatch(str string) string {
 	p.t.Helper()
 
-	timeout, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	timeout, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
```
Comment (AbhineetJain) [inline pty/ptytest/ptytest.go:89]: I see some instances where the wait times have been updated. For example, if we were mapping directly, we would use ‘WaitMedium’ here?

**Review by AbhineetJain — APPROVED** [review]
Just have a small question, but this is an awesome PR! 🔥

File: pty/ptytest/ptytest.go (hunk 3 above)
Comment (mafredri) [reply pty/ptytest/ptytest.go:89]: `WaitMedium` was a later addition, so this was a good catch!

**Review by kylecarbs — APPROVED** [review]
I'm a fan of `testtime` or `testdir` more.

`util` packages expand infinitely, I'm convinced!

**Comment by mafredri** [comment]
I moved the `testutil` package from `/internal` to `/` based on team feedback. I'll leave it as `testutil` for now, it's a distinct name so changing it in the future should be quite easy if we find a better fit.

**Pull request merged**